core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -107,7 +107,8 @@ private void registerWithMaster() throws IOException {
         ConfigurationUtils.getConfiguration(ServerConfiguration.global(), Scope.WORKER);
     mMasterClient.register(mWorkerId.get(),
         storageTierAssoc.getOrderedStorageAliases(), storeMeta.getCapacityBytesOnTiers(),
-        storeMeta.getUsedBytesOnTiers(), storeMeta.getBlockList(), configList);
+        storeMeta.getUsedBytesOnTiers(), storeMeta.getBlockList(),
+        storeMeta.getLostStorage(), configList);
```

Comment:
Will this parameter include all lost storage paths including the new ones reported in heartbeat with the previous active master?

Reply:
The `LostStorageIntegrationTest.lostStorageWhenRestart()` is designed to make sure that the lost storage information in the previous master can sustain through master restarts. This parameter include all lost storage paths that this worker failed to initialize or lost (which are reported to the previous masters)

Reply:
I see. Thanks for the explanation.